clients/banking/src/components/CardItemSettings.tsx

```diff
@@ -159,6 +169,44 @@ export const CardItemSettings = ({
         canManageCards={canManageCards}
       />
 
+      {match(accountHolder)
+        .with({ info: { type: "Company" } }, () => {
+          const country = getCCA2forCCA3(card.issuingCountry as CountryCCA3);
+          if (isNotNullishOrEmpty(country)) {
+            return (
+              <>
+                <Space height={24} />
+
+                <LakeText variant="smallRegular">
+                  {formatNestedMessage("card.mastercardBonusProgramLink", {
+                    learnMoreLink: (
+                      <>
+                        <Link
+                          style={styles.link}
+                          to={`https://www.mastercard.com/businessbonus/${country.toLowerCase()}/home`}
+                          target="blank"
+                        >
+                          <Box direction="row" alignItems="center">
+                            <LakeText color={colors.current.primary} variant="smallRegular">
+                              {t("common.learnMore")}
+                            </LakeText>
+
+                            <Space width={4} />
+                            <Icon color={colors.current.primary} name="open-filled" size={16} />
+                          </Box>
+                        </Link>
+                      </>
+                    ),
+                  })}
+                </LakeText>
+
+                <Space height={16} />
+              </>
+            );
+          }
+        })
+        .otherwise(() => null)}
```

Comment:
By avoiding the cast, you can put all your condition in the match and it improves the logic readability as it's in one place:

```tsx
{match({
  type: accountHolder?.info.type,
  country: countries.find(country => country.cca3 === card.issuingCountry),
})
  .with({ type: "Company", country: P.not(P.nullish) }, ({ country }) => (
    <>
      <Space height={24} />

      <LakeText variant="smallRegular">
        {formatNestedMessage("card.mastercardBonusProgramLink", {
          learnMoreLink: (
            <>
              <Link
                style={styles.link}
                to={`https://www.mastercard.com/businessbonus/${country.cca2.toLowerCase()}/home`}
                target="blank"
              >
                <Box direction="row" alignItems="center">
                  <LakeText color={colors.current.primary} variant="smallRegular">
                    {t("common.learnMore")}
                  </LakeText>

                  <Space width={4} />
                  <Icon color={colors.current.primary} name="open-filled" size={16} />
                </Box>
              </Link>
            </>
          ),
        })}
      </LakeText>

      <Space height={16} />
    </>
  ))
  .otherwise(() => null)}
```

Or, if your prefer:

```tsx
{match({
  type: accountHolder?.info.type,
  country: isCountryCCA3(card.issuingCountry)
    ? getCCA2forCCA3(card.issuingCountry)?.toLowerCase()
    : undefined,
})
  .with({ type: "Company", country: P.not(P.nullish) }, ({ country }) => (
    <>
      <Space height={24} />

      <LakeText variant="smallRegular">
        {formatNestedMessage("card.mastercardBonusProgramLink", {
          learnMoreLink: (
            <>
              <Link
                style={styles.link}
                to={`https://www.mastercard.com/businessbonus/${country}/home`}
                target="blank"
              >
                <Box direction="row" alignItems="center">
                  <LakeText color={colors.current.primary} variant="smallRegular">
                    {t("common.learnMore")}
                  </LakeText>

                  <Space width={4} />
                  <Icon color={colors.current.primary} name="open-filled" size={16} />
                </Box>
              </Link>
            </>
          ),
        })}
      </LakeText>

      <Space height={16} />
    </>
  ))
  .otherwise(() => null)}
```

Otherwise, LGTM